.github/workflows/main-java.yml

```
@@ -0,0 +1,48 @@
+name: Build and Test (Java)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+  merge_group:
+    paths:
+      - 'pkg/java/**'
+      - 'OpenFGAParser.g4'
+      - 'OpenFGALexer.g4'
+      - 'tests'
+  pull_request:
+    paths:
+      - 'pkg/java/**'
+      - 'OpenFGAParser.g4'
+      - 'OpenFGALexer.g4'
+      - 'tests'
+
+permissions:
+  contents: read
+
+jobs:
+
+    build:
+      name: Build and Test Java
+      runs-on: ubuntu-latest
+
+      strategy:
+        matrix:
+          java: [ '11', '17', '20' ]
+
+      steps:
+        - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v3.5.2
```

```suggestion
        - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
          with:
            fetch-depth: 0

        - name: Set up Docker Buildx
          uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
```

This is needed to get CI to run